Task10_Auth/src/test/java/CreateUsersTests.java

```diff
@@ -0,0 +1,106 @@
+import java.util.List;
+import java.util.Random;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class CreateUsersTests {
+    private ZipCodesClient zipCodesClient = new ZipCodesClient();
+    private UsersClient usersClient = new UsersClient();
+    private static final Random random = new Random();
+
+    @Test
+    public void createUserTest() {
+        ZipCodesClient.createZipCodeIfNotExist(zipCodesClient);
+        ResponseWrapper getZipCodesResponseBeforeCreation = zipCodesClient.getZipCodes();
+        int zipCodesSize = getZipCodesResponseBeforeCreation.getResponseBody().size();
+        List<String> zipCodesListBeforeCreation = new ArrayList<>(getZipCodesResponseBeforeCreation.getResponseBody());
+        String exactZipCode = zipCodesListBeforeCreation.get(random.nextInt(zipCodesSize));
+        UsersDto userToAdd = new UsersDto();
+        userToAdd.setZipCode(exactZipCode);
+
+        ResponseWrapper createUserResponse = usersClient.createUser(userToAdd);
+
+        ResponseWrapper getZipCodesResponseAfterCreation = zipCodesClient.getZipCodes();
+        List<String> zipCodesListAfterCreation = new ArrayList<>(getZipCodesResponseAfterCreation.getResponseBody());
+        ResponseWrapper getUsersResponse = usersClient.getUsers();
+        List<UsersDto> usersList = new ArrayList<>(getUsersResponse.getResponseBody());
+        UsersDto addedUser = usersList.get(usersList.indexOf(userToAdd));
+        assertAll("Status code is 201 // user is added, zip code is removed",
+                () -> assertFalse(zipCodesListAfterCreation.contains(exactZipCode)),
+                () -> assertTrue(usersList.contains(userToAdd)),
+                () -> assertEquals(userToAdd, addedUser),
+                () -> assertEquals(201, createUserResponse.getResponseCode()));
+    }
+
+    @Test
+    public void createUserWithRequiredFieldsTest() {
+        String name = RandomStringUtils.random(5, true, false);
+        UsersDto.Sex userSex = UsersDto.Sex.getRandomSex();
+        UsersDto userToAdd = new UsersDto(name, userSex);
+
+        ResponseWrapper createUserResponse = usersClient.createUser(userToAdd);
+
+        ResponseWrapper getUsersResponse = usersClient.getUsers();
+        List<UsersDto> usersList = new ArrayList<>(getUsersResponse.getResponseBody());
+        UsersDto addedUser = usersList.get(usersList.indexOf(userToAdd));
+        assertAll("Status code is 201 // user is added",
+                () -> assertTrue(usersList.contains(userToAdd)),
+                () -> assertEquals(userToAdd, addedUser),
+                () -> assertEquals(201, createUserResponse.getResponseCode()));
+    }
+
+    @Test
+    public void createUserWithIncorrectZipCodeTest() {
+        ResponseWrapper getZipCodesResponse = zipCodesClient.getZipCodes();
+        String zipCode;
+        do {
+            zipCode = RandomStringUtils.random(5, false, true);
+        } while (getZipCodesResponse.getResponseBody().contains(zipCode));
```

Comment:
I think you can create ZipCodeGenerator class and paste these code in method in it, in this case you will encapsulate this code:
`ZipCodeGenerator.getRandomZipCode()`

Reply:
done

I've moved it to ZipCodesClient instead of a separate class.